src/sdk/object_sdk.js

```diff
@@ -79,6 +82,25 @@ const MULTIPART_NAMESPACES = [
 ];
 const required_obj_properties = ['obj_id', 'bucket', 'key', 'size', 'content_type', 'etag'];
 
+/** validation function that is used as a callback from all the services:
+ * S3 (object_sdk), IAM (account_sdk), STS (sts_sdk)
+ * @param {object} data
+ * @param {object} params
+ */
+async function _validate_account(data, params) {
+    const time = Date.now();
+    // stat check (only in bucketspace FS)
+    const bs = params.bucketspace;
+    const bs_allow_stat_account = Boolean(bs.check_same_stat_account);
+    if (bs_allow_stat_account && config.NC_ENABLE_ACCOUNT_CACHE_STAT_VALIDATION) {
+        const same_stat = await bs.check_same_stat_account(data._id, data.stat);
+        if (!same_stat) { // config file of bucket was changed
+            return false;
+        }
+    }
+    if (time <= data.valid_until) return true;
```

Comment:
Are you using here the valid_until of the bucketspace?

Reply:
You're were right, I missed to set it and it was always `undefined`, now it is set it in the phase where we call the `account_cache.get_with_cache`, since it is configured to 10 minutes so I'm fine with it called prior to the actual load.